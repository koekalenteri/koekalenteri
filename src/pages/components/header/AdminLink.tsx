import { useTranslation } from 'react-i18next'
import { Link as SpaLink } from 'react-router-dom'
import Link from '@mui/material/Link'
import Typography from '@mui/material/Typography'

import { Path } from '../../../routeConfig'

interface Props {
  readonly active?: boolean
  readonly activeBorder?: string
}

export const AdminLink = ({ active, activeBorder }: Props) => {
  const { t } = useTranslation()
  const border = active ? activeBorder : undefined

  return (
    <Link
      to={Path.admin.index}
      component={SpaLink}
      sx={{
        textDecoration: 'none',
        borderBottom: border,
        mr: 1,
        px: 1,
      }}
    >
      <Typography
        variant="subtitle1"
        noWrap
        component="div"
        sx={{
          flexShrink: 1,
        }}
      >
        {t('admin')}
      </Typography>
    </Link>
  )
}
