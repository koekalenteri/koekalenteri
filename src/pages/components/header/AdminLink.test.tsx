import type { ReactNode } from 'react'

import { MemoryRouter } from 'react-router'
import { render, screen } from '@testing-library/react'

import { AdminLink } from './AdminLink'

const Wrapper = (props: { readonly children?: ReactNode }) => {
  return <MemoryRouter>{props.children}</MemoryRouter>
}
describe('AdminLink', () => {
  it('should render with minimal properties', () => {
    const { container } = render(<AdminLink />, { wrapper: Wrapper })
    expect(container).toMatchSnapshot()

    const link = screen.getByRole('link', { name: 'admin' })
    expect(link).toHaveAttribute('href', '/admin/event')
  })

  it('should render active border', () => {
    const { container } = render(<AdminLink active activeBorder="1px solid red" />, { wrapper: Wrapper })
    expect(container).toMatchSnapshot()

    const link = screen.getByRole('link', { name: 'admin' })
    expect(link).toHaveStyle({ borderBottom: '1px solid red' })
  })

  it('should not render border when not active', () => {
    const { container } = render(<AdminLink activeBorder="1px solid red" />, { wrapper: Wrapper })
    expect(container).toMatchSnapshot()

    const link = screen.getByRole('link', { name: 'admin' })
    expect(link).not.toHaveStyle({ borderBottom: '1px solid red' })
  })
})
