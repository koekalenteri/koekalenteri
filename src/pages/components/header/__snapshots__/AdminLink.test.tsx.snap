// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminLink should not render border when not active 1`] = `
<div>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eum0yc-MuiTypography-root-MuiLink-root"
    href="/admin/event"
  >
    <div
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-noWrap css-vq2sim-MuiTypography-root"
    >
      admin
    </div>
  </a>
</div>
`;

exports[`AdminLink should render active border 1`] = `
<div>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1m4q2di-MuiTypography-root-MuiLink-root"
    href="/admin/event"
  >
    <div
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-noWrap css-vq2sim-MuiTypography-root"
    >
      admin
    </div>
  </a>
</div>
`;

exports[`AdminLink should render with minimal properties 1`] = `
<div>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-eum0yc-MuiTypography-root-MuiLink-root"
    href="/admin/event"
  >
    <div
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-noWrap css-vq2sim-MuiTypography-root"
    >
      admin
    </div>
  </a>
</div>
`;
